### ISSUE TITLE
Assertion 'path' failed .. Freezing execution.

### ISSUE THREAD

**Issue by mbiebl** [systemd/systemd#5441]
### Submission type

  - [X] Bug report
  - [ ] Request for enhancement (RFE)

### systemd version the issue has been seen with

215-17+deb8u6
232-18

### Used distribution

Debian stretch
Debian jessie

See the downstream bug report https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=856035

Hi, on jessie (amd64 kvm/qemu VM) I can reliably trigger this:

```
systemd[1]: Got automount request for (null), triggered by 1767 (ls)
systemd[1]: Assertion 'path' failed at ../src/shared/mkdir.c:79, function mkdir_parents_internal(). Aborting.
systemd[1]: Caught <ABRT>, dumped core as pid 1768.
systemd[1]: Freezing execution.
```

assertion error as follows:
(Ensure proc-sys-fs-binfmt_misc.automount is not masked)

```
### disable module loading
echo 1 > /proc/sys/kernel/modules_disabled
 
### trigger binfmt automount
ls -l /proc/sys/fs/binfmt_misc/

### check that proc-sys-fs-binfmt_misc.mount has failed
systemctl --failed

### mask the automounter
systemctl mask proc-sys-fs-binfmt_misc.automount

### tail the jornal (optional)
journalctl -f &

### trigger the assertion error
ls -l /proc/sys/fs/binfmt_misc/
```
This came up because we mask proc-sys-fs-binfmt_misc.automount using
puppet and then systemd froze on all our hosts and things were very sad
until the world got rebooted.

**Comment by mbiebl** [comment]
This is also reproducible with git master (ecc0eab247da25a6767ccabd2162a4d03de6ee8c)

**Comment by mbiebl** [comment]
An interesting side-effect: if you have systemd-coredump installed and enabled, it will segfault:
```
[  155.306761] systemd-coredump[744]: Due to PID 1 having crashed coredump collection will now be turned off.
[  155.306805] systemd-coredum[744]: segfault at 0 ip 00007fcbe0dd3486 sp 00007ffd900b9c68 error 4 in libc-2.24.so[7fcbe0d53000+195000]
[  155.307647] systemd: 36 output lines suppressed due to ratelimiting
```

**Comment by mbiebl** [comment]
Filed the systemd-coredump segfault as a separate issue, see #5442

**Comment by mbiebl** [comment]
```
#0  0x00007f308a8ee2f7 in kill () at ../sysdeps/unix/syscall-template.S:84
No locals.
#1  0x0000557b588c166a in crash.lto_priv.649 (sig=6) at ../src/core/main.c:190
        sa = {__sigaction_handler = {sa_handler = 0x0, sa_sigaction = 0x0}, 
          sa_mask = {__val = {0 <repeats 16 times>}}, sa_flags = 0, 
          sa_restorer = 0x0}
        __func__ = "crash"
        __PRETTY_FUNCTION__ = "crash"
#2  <signal handler called>
No locals.
#3  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:58
        set = {__val = {8927191084203854339, 26, 0, 139846484596382, 0, 0, 0, 
            139846485818784, 139846484881741, 61, 139846485007904, 0, 0, 0, 0, 
            140728783724160}}
        pid = <optimized out>
        tid = <optimized out>
#4  0x00007f308a8ef40a in __GI_abort () at abort.c:89
        save_stage = 2
        act = {__sigaction_handler = {sa_handler = 0x557b5a713618, 
            sa_sigaction = 0x557b5a713618}, sa_mask = {__val = {
              93988255031264, 140728783724352, 16222923957520311296, 
              206158430256, 139846484881741, 61, 139846485007904, 2, 0, 0, 
              139846484598371, 0, 0, 0, 139846485818784, 12884901887}}, 
          sa_flags = 0, sa_restorer = 0x7ffdf929e040}
Quit
#0  0x00007f308a8ee2f7 in kill () at ../sysdeps/unix/syscall-template.S:84
No locals.
#1  0x0000557b588c166a in crash.lto_priv.649 (sig=6) at ../src/core/main.c:190
        sa = {__sigaction_handler = {sa_handler = 0x0, sa_sigaction = 0x0}, 
          sa_mask = {__val = {0 <repeats 16 times>}}, sa_flags = 0, 
          sa_restorer = 0x0}
        __func__ = "crash"
        __PRETTY_FUNCTION__ = "crash"
#2  <signal handler called>
No locals.
#3  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:58
        set = {__val = {8927191084203854339, 26, 0, 139846484596382, 0, 0, 0, 
            139846485818784, 139846484881741, 61, 139846485007904, 0, 0, 0, 0, 
            140728783724160}}
        pid = <optimized out>
        tid = <optimized out>
#4  0x00007f308a8ef40a in __GI_abort () at abort.c:89
        save_stage = 2
        act = {__sigaction_handler = {sa_handler = 0x557b5a713618, 
            sa_sigaction = 0x557b5a713618}, sa_mask = {__val = {
              93988255031264, 140728783724352, 16222923957520311296, 
              206158430256, 139846484881741, 61, 139846485007904, 2, 0, 0, 
              139846484598371, 0, 0, 0, 139846485818784, 12884901887}}, 
          sa_flags = 0, sa_restorer = 0x7ffdf929e040}
        sigs = {__val = {32, 0 <repeats 15 times>}}
#5  0x00007f308c09a1f2 in log_assert_failed (text=<optimized out>, 
    file=<optimized out>, line=<optimized out>, func=<optimized out>)
    at ../src/basic/log.c:796
No locals.
#6  0x00007f308c07f3ef in mkdir_parents_internal (prefix=<optimized out>, 
    path=path@entry=0x0, mode=mode@entry=493, 
    _mkdir=_mkdir@entry=0x7f308c063100 <mkdir_label>)
    at ../src/basic/mkdir.c:61
        p = <optimized out>
        e = <optimized out>
        r = <optimized out>
        __PRETTY_FUNCTION__ = "mkdir_parents_internal"
#7  0x00007f308c07f422 in mkdir_p_internal (prefix=<optimized out>, 
    path=<optimized out>, mode=<optimized out>, _mkdir=<optimized out>, 
    path=<optimized out>) at ../src/basic/mkdir.c:115
        r = <optimized out>
#8  0x0000557b588bbf60 in automount_enter_runnning (a=0x557b5a7132e0)
    at ../src/core/automount.c:745
        __func__ = "automount_enter_runnning"
        error = {name = 0x0, message = 0x0, _need_free = 0}
        st = {st_dev = 93988286607888, st_ino = 139846483911338, 
          st_nlink = 93988287543168, st_mode = 1517551664, st_uid = 21883, 
          st_gid = 1517269520, __pad0 = 21883, st_rdev = 139846484666958, 
          st_size = 93986769338369, st_blksize = 93988286607888, 
          st_blocks = 0, st_atim = {tv_sec = 139846484667051, 
            tv_nsec = 93988286890032}, st_mtim = {tv_sec = 0, 
            tv_nsec = 93988286607888}, st_ctim = {tv_sec = 0, tv_nsec = 0}, 
          __glibc_reserved = {93988286607888, 1, 139846484040457}}
        r = <optimized out>
        __func__ = "automount_enter_runnning"
#9  automount_dispatch_io.lto_priv.45 (s=<optimized out>, fd=<optimized out>, 
    events=<optimized out>, userdata=0x557b5a7132e0)
    at ../src/core/automount.c:993
        error = {name = 0x0, message = 0x0, _need_free = 0}
        packet = {hdr = {proto_version = 5, type = 5}, v5_packet = {hdr = {
              proto_version = 5, type = 5}, wait_queue_token = 2, dev = 34, 
            ino = 8904, uid = 0, gid = 0, pid = 487, tgid = 487, len = 16, 
            name = "ffff96f12c934a80", '\000' <repeats 239 times>}, 
          missing_indirect = {hdr = {proto_version = 5, type = 5}, 
            wait_queue_token = 2, dev = 34, ino = 8904, uid = 0, gid = 0, 
            pid = 487, tgid = 487, len = 16, 
            name = "ffff96f12c934a80", '\000' <repeats 239 times>}, 
          expire_indirect = {hdr = {proto_version = 5, type = 5}, 
            wait_queue_token = 2, dev = 34, ino = 8904, uid = 0, gid = 0, 
            pid = 487, tgid = 487, len = 16, 
            name = "ffff96f12c934a80", '\000' <repeats 239 times>}, 
          missing_direct = {hdr = {proto_version = 5, type = 5}, 
            wait_queue_token = 2, dev = 34, ino = 8904, uid = 0, gid = 0, 
            pid = 487, tgid = 487, len = 16, 
            name = "ffff96f12c934a80", '\000' <repeats 239 times>}, 
          expire_direct = {hdr = {proto_version = 5, type = 5}, 
            wait_queue_token = 2, dev = 34, ino = 8904, uid = 0, gid = 0, 
            pid = 487, tgid = 487, len = 16, 
            name = "ffff96f12c934a80", '\000' <repeats 239 times>}}
        st = {st_dev = 93988286607888, st_ino = 139846483911338, 
          st_nlink = 93988287543168, st_mode = 1517551664, st_uid = 21883, 
          st_gid = 1517269520, __pad0 = 21883, st_rdev = 139846484666958, 
          st_size = 93986769338369, st_blksize = 93988286607888, 
          st_blocks = 0, st_atim = {tv_sec = 139846484667051, 
            tv_nsec = 93988286890032}, st_mtim = {tv_sec = 0, 
            tv_nsec = 93988286607888}, st_ctim = {tv_sec = 0, tv_nsec = 0}, 
          __glibc_reserved = {93988286607888, 1, 139846484040457}}
        r = <optimized out>
        __PRETTY_FUNCTION__ = "automount_dispatch_io"
        __func__ = "automount_dispatch_io"
#10 0x00007f308c0863d6 in source_dispatch (s=s@entry=0x557b5a6fe6a0)
    at ../src/libsystemd/sd-event/sd-event.c:2272
        saved_type = <optimized out>
        r = <optimized out>
        __PRETTY_FUNCTION__ = "source_dispatch"
        __func__ = "source_dispatch"
#11 0x00007f308c0865a4 in sd_event_dispatch (e=e@entry=0x557b5a6f0080)
    at ../src/libsystemd/sd-event/sd-event.c:2631
        p = <optimized out>
        r = <optimized out>
        __PRETTY_FUNCTION__ = "sd_event_dispatch"
#12 0x00007f308c087ad7 in sd_event_run (e=0x557b5a6f0080, 
    timeout=18446744073709551615) at ../src/libsystemd/sd-event/sd-event.c:2690
        r = 1
        __PRETTY_FUNCTION__ = "sd_event_run"
#13 0x0000557b58892756 in manager_loop (m=0x557b5a6eea00)
    at ../src/core/manager.c:2284
        wait_usec = <optimized out>
        r = <optimized out>
        rl = {interval = 1000000, begin = 35373687, burst = 50000, num = 8}
        __PRETTY_FUNCTION__ = "manager_loop"
        __func__ = "manager_loop"
#14 0x0000557b5884925a in main (argc=<optimized out>, argv=<optimized out>)
    at ../src/core/main.c:1914
        m = 0x557b5a6eea00
        r = <optimized out>
        retval = 1
        before_startup = 3723068
        after_startup = <optimized out>
        timespan = "8\347)\371\375\177\000\000H\347)\371\375\177\000\000\350<\371\213\060\177\000\000\000\000\000\000\000\000\000\000\001\000\000\000\000\000\000\000DJ\035\214\060\177\000\000\001", '\000' <repeats 14 times>
        fds = 0x0
        reexecute = false
        shutdown_verb = 0x0
        initrd_timestamp = <optimized out>
        userspace_timestamp = {realtime = 1487953607850843, 
          monotonic = 3697011}
        kernel_timestamp = {realtime = 1487953604153832, monotonic = 0}
        security_start_timestamp = {realtime = 1487953607852277, 
          monotonic = 3698446}
        security_finish_timestamp = {realtime = 1487953607852399, 
          monotonic = 3698567}
        systemd = "systemd"
        skip_setup = <optimized out>
        j = <optimized out>
        loaded_policy = false
        arm_reboot_watchdog = false
        queue_default_job = true
        empty_etc = <optimized out>
        switch_root_dir = 0x0
        switch_root_init = 0x0
        saved_rlimit_nofile = {rlim_cur = 1024, rlim_max = 4096}
        error_message = 0x0
        __func__ = "main"
        __PRETTY_FUNCTION__ = "main"
```

**Comment by poettering** [comment]
Fix waiting in #5445.
